cmd/hydrate/src/lib.rs

```diff
@@ -0,0 +1,217 @@
+            bail!(concat!(
+                "DryCore does not support ",
+                stringify!($fn_name)))
+        }
+    };
+}
+
+impl humility::core::Core for DryCore {
+    unsupported!(run());
+    unsupported!(halt());
+    unsupported!(step());
+    unsupported!(load(_path: &std::path::Path));
+    unsupported!(reset());
+    unsupported!(write_8(_addr: u32, _data: &[u8]));
+    unsupported!(op_done());
+    unsupported!(op_start());
+    unsupported!(read_reg(_reg: ARMRegister) -> Result<u32>);
+    unsupported!(init_swv());
+    unsupported!(read_swv() -> Result<Vec<u8>>);
+    unsupported!(write_reg(_reg: ARMRegister, _value: u32));
+    unsupported!(write_word_32(_addr: u32, _data: u32));
+    unsupported!(reset_and_halt(_dur: std::time::Duration));
+    unsupported!(wait_for_halt(_dur: std::time::Duration));
+
+    fn info(&self) -> (String, Option<String>) {
+        ("DryCore".to_owned(), None)
+    }
+    fn is_archive(&self) -> bool {
+        false
+    }
+    fn is_dump(&self) -> bool {
+        true // I guess?
+    }
+    fn is_net(&self) -> bool {
+        false
+    }
+    fn vid_pid(&self) -> Option<(u16, u16)> {
+        None
+    }
+
+    fn read_8(&mut self, addr: u32, data: &mut [u8]) -> Result<()> {
+        if self.flash.read(addr, data).is_some() {
+            return Ok(());
+        }
+
+        let Some((base, mem)) = self.mem.range(0..=addr).next_back() else {
+            bail!("addr {addr:#08x} is below memory range");
+        };
+        let offset = (addr - base) as usize;
+        let end = offset + data.len();
+        if end > mem.len() {
+            bail!("region is not large enough; {end:#x} > {:#x}", mem.len());
+        }
+        data.copy_from_slice(&mem[offset..end]);
+        Ok(())
+    }
+}
+
+fn run(context: &mut ExecutionContext) -> Result<()> {
+    let Subcommand::Other(subargs) = context.cli.cmd.as_ref().unwrap();
+    let subargs = Args::try_parse_from(subargs)?;
+    let f = std::fs::File::open(&subargs.file)?;
+    let mut z = zip::ZipArchive::new(f)?;
+
+    let mut s = String::new();
+    z.by_name("meta.json")?.read_to_string(&mut s)?;
+    #[derive(serde::Deserialize)]
+    struct Meta {
+        version: u64,
+    }
+    let meta: Meta = serde_json::from_str(&s)?;
+    if meta.version != 1 {
+        bail!(
+            "invalid version in `meta.json`: expected 1, got {}",
+            meta.version
+        );
+    }
+
+    let mut s = String::new();
+    z.by_name("TASK_INDEX")?.read_to_string(&mut s)?;
+    let task_id: u16 = s.trim().parse()?;
+
+    let mut s = String::new();
+    z.by_name("TIMESTAMP")?.read_to_string(&mut s)?;
+    let timestamp: u64 = s.trim().parse()?;
+
+    let mut bord = String::new();
+    z.by_name("BORD")?.read_to_string(&mut bord)?;
+    bord.pop(); // remove newline
+
+    let mut gitc = String::new();
+    z.by_name("GITC")?.read_to_string(&mut gitc)?;
+    gitc.pop(); // remove newline
```

Comment:
nit, take it or leave it: perhaps we ought to make this explicitly remove _only_ a newline, rather than "whatever the last character is"? i realize these are always expected to end with a newline, but it might be nice to be forward compatible and handle any potential future changes where they're _not_ newline-terminated, rather than just happily corrupting the commit SHA or board name if the newline ever goes away?

```suggestion
    let bord = bord.trim_end_matches('\n'); // remove newline

    let mut gitc = String::new();
    z.by_name("GITC")?.read_to_string(&mut gitc)?;
    let gitc = gitc.trim_end_matches('\n');; // remove newline
```